homeassistant/components/tis_control/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The TISControl integration."""
+
+from __future__ import annotations
+
+import logging
+
+from attr import dataclass
+from TISControlProtocol.api import TISApi
+from TISControlProtocol.Protocols.udp.ProtocolHandler import TISProtocolHandler
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DEVICES_DICT, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+type TISConfigEntry = ConfigEntry[TISData]
+protocol_handler = TISProtocolHandler()
+
+
+@dataclass
+class TISData:
+    """TISControl data stored in the ConfigEntry."""
+
+    api: TISApi
```

Comment:
```suggestion
type TISConfigEntry = ConfigEntry[TISApi]
protocol_handler = TISProtocolHandler()
```
just type it with the API, no need for the abstraction